src/use-swr.ts

```diff
@@ -306,8 +306,9 @@ export const useSWRHandler = <Data = any, Error = any>(
   // Similar to the global mutate, but bound to the current cache and key.
   // `cache` isn't allowed to change during the lifecycle.
   const boundMutate: SWRResponse<Data, Error>['mutate'] = useCallback(
-    (newData, shouldRevalidate) =>
-      internalMutate(cache, keyRef.current, newData, shouldRevalidate),
+    // By using `bind` we don't need to modify the size of the rest arguments.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
lint warning:
```
src/use-swr.ts#L315
React Hook useCallback received a function whose dependencies are unknown. Pass an inline function instead
```

guess you need to put it before line of deps.

We had issue with bind functions before, will it still have the similar issue then?

Reply:
> We had issue with bind functions before, will it still have the similar issue then?

It should be OK since here we're always referring to the latest key. We've also added test for it.